### PR TITLE
cleaned the beam code a bit

### PR DESCRIPTION
**Pull request by KlausBSautter**
did some cleaning

### CODE REVIEW

**Review by loumalouomega — APPROVED** [review]
It looks fine

**Comment by philbucher** [comment]
@KlausBSautter you might have a look at the `Length` function of the Geometry. You might be able use it in your `CalculateReferenceLength` and `CalculateCurrentLength` methods

**Comment by RiccardoRossi** [comment]
Hi Guys,

i can not review this now. I trust you...

On a side note, please verify if the new interface works correctly, and IF THE LOCAL AXES provided from the interface work correctly in the tests

File: applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.hpp [hunk 1]
```diff
@@ -191,10 +192,9 @@ namespace Kratos
...
 	private:
-		double mdPhi_x_a, mRotInertiaY, mRotInertiaZ;
-		Vector mNX, mNY, mNZ, mRHS, mTotalDef, mTotalPos;
-		Vector mTotalNodalDeformation, mTotalNodalPosistion, mBodyForces;
-		Vector mDeformationModes, mIncrementDeformation;
+		Vector mNX, mNY, mNZ;
+		Vector mTotalNodalDeformation;
+		Vector mIncrementDeformation;
 		Matrix mLHS, mRotationMatrix;
 		bounded_matrix<double,msElementSize,msElementSize> mRotationMatrix0;
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.hpp:198]: why do you store the mLHS, the mRotationMatrix and the mRotationMatrix0?

if you have the quaterninon you can compute them efficiently when needed

**Review by RiccardoRossi — COMMENTED** [review]
I just had a quick look.

I am convinced you could avoid storing most of the variables that you store in the class.

are u really sure you can not live without

**Pull request merged**